src/Core/ISolutionInfoProvider.cs

```diff
@@ -77,5 +77,16 @@ public interface ISolutionInfoProvider
         /// </summary>
         /// <remarks>to be used only by existing non-thread-aware code that cannot easily be refactored.</remarks>
         bool IsFolderWorkspace();
+
+        /// <summary>
+        /// Returns the Solution Name, or null if there is not a current solution
+        /// </summary>
+        Task<string> GetSolutionNameAsync();
```

Comment:
UnintrusiveBindingPathProvider.GetConnectionFilePath can be changed to use this method as well